include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,370 @@
+
+  //! Markup to trigger construction of container without initialisation of content
+  inline constexpr no_init_t no_init = {};
+
+  //================================================================================================
+  //! @addtogroup memory
+  //! @{
+  //!   @brief SIMD-aware container for product types
+  //!
+  //!   **Required header:** `#include <eve/memory/soa_vector.hpp>`
+  //!
+  //!   soa_vector is a container similar to `std::vector` but optimized to store
+  //!   [product types](eve::product_type) using a Structure Of Array layout compatible with SIMD
+  //!   processing.
+  //!
+  //!   @tparam Type  Value type stored inside `eve::soa_vector`
+  //! @}
+  //================================================================================================
+  template< eve::product_type Type
+          , eve::simd_allocator Allocator = eve::aligned_allocator<unsigned char, eve::fixed<64>>
+          >
+  struct soa_vector
+  {
+    public:
+
+    using storage_t = detail::soa_storage<Type,Allocator>;
+
+    //!=============================================================================================
+    //! @name Member types
+    //! @{
+    //==============================================================================================
+    //! Type of data stored in the container
+    using value_type = Type;
+
+    //!   @brief pointer and iterator types.
+    //!   pointer*  - are eve::algo::views::zip_iterator over fields (no conversion to type, just flat fields)
+    //!   iterator* - are a pointer, converter to the Type.
+    //!   They all satisfy eve::algo::relaxed_iterator but not std::iterator
+
+    using pointer               = decltype(storage_t{}.data());
+    using const_pointer         = decltype(std::declval<storage_t const>().data());
+    using pointer_aligned       = decltype(storage_t{}.data_aligned());
+    using const_pointer_aligned = decltype(std::declval<storage_t const>().data_aligned());
+
+    using iterator               = decltype(views::convert(pointer{}              , as<value_type>{}));
+    using const_iterator         = decltype(views::convert(const_pointer{}        , as<value_type>{}));
+    using iterator_aligned       = decltype(views::convert(pointer_aligned{}      , as<value_type>{}));
+    using const_iterator_aligned = decltype(views::convert(const_pointer_aligned{}, as<value_type>{}));
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Constructors
+    //! @{
+    //==============================================================================================
+
+    //! Constructs an empty container.
+    explicit soa_vector(Allocator a = Allocator{}) : storage(a), size_{} {}
```

Comment:
And yeah - I think this will sfinae out if your allocator is not default constructible.
So - I think  `=Allocator {}` won't fly in all constructors.

Reply:
Fixed